### PR TITLE
Mutable trigger levels in PriceAuthority

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
In service of #1947

### CODE REVIEW

File: packages/zoe/src/contractSupport/priceAuthority.js [hunk 1]
```diff
@@ -138,6 +141,64 @@ export function makeOnewayPriceAuthorityKit(opts) {
...
+      let amountIn = amountMath.coerce(amountInArg, actualBrandIn);
+      let amountOutLimit = amountMath.coerce(amountOutLimitArg, actualBrandOut);
+
+      /** @type {PromiseRecord<PriceQuote>} */
+      const triggerPK = makePromiseKit();
+
+      const mutableQuote = Far('MutableQuote', {
+        cancel: e => triggerPK.reject(e),
+        updateLevel: (newAmountIn, newAmountOutLimit) => {
+          amountIn = amountMath.coerce(newAmountIn, actualBrandIn);
```
Comment (michaelfig) [inline packages/zoe/src/contractSupport/priceAuthority.js]: Can you handle when `newAmountIn` succeeds in coercing but `newAmountOutLimit` fails the coercion?

File: packages/zoe/test/unitTests/contracts/test-priceAggregator.js [hunk 2]
```diff
@@ -476,3 +476,230 @@ test('quoteWhen', /** @param {ExecutionContext} t */ async t => {
...
+  );
+
+  await E(oracleTimer).tick();
+  // Above trigger has not yet fired.
+  t.falsy(abovePriceQuote);
+  await E(oracleTimer).tick();
+
+  // The above trigger should fire here.
+  await E(mutableQuoteWhenGTE).getPromise();
+  t.truthy(abovePriceQuote);
```
Comment (michaelfig) [inline packages/zoe/test/unitTests/contracts/test-priceAggregator.js]:
```suggestion
  t.truthy(abovePriceQuote);
  await E(mutableQuoteWhenGTE).getPromise();
```

File: packages/zoe/test/unitTests/contracts/test-priceAggregator.js [hunk 3]
```diff
@@ -476,3 +476,230 @@ test('quoteWhen', /** @param {ExecutionContext} t */ async t => {
...
+  await E(oracleTimer).tick();
+  await E(oracleTimer).tick();
+
+  // Below trigger has not yet fired.
+  t.falsy(belowPriceQuote);
+  await E(price1300Admin).delete();
+
+  // The below trigger should fire here.
+  await E(mutableQuoteWhenLTE).getPromise();
+  t.truthy(belowPriceQuote);
```
Comment (michaelfig) [inline packages/zoe/test/unitTests/contracts/test-priceAggregator.js:587]:
```suggestion
  t.truthy(belowPriceQuote);
  await E(mutableQuoteWhenLTE).getPromise();
```

**Review by michaelfig — APPROVED** [review]
Conditional approval: please address the comments when you have a chance.

File: packages/zoe/src/contractSupport/priceAuthority.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/contractSupport/priceAuthority.js]: done

File: packages/zoe/test/unitTests/contracts/test-priceAggregator.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-priceAggregator.js]: done

File: packages/zoe/test/unitTests/contracts/test-priceAggregator.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-priceAggregator.js:587]: This change doesn't pass. If I understand the semantics of `async` correctly, the problem is that `delete()` doesn't wait for the `updateState()`  in `updateQuote()` to propagate.

**Pull request merged**